core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -87,7 +89,12 @@ private[spark] class CoarseMesosSchedulerBackend(
 
   override def start() {
     super.start()
-
+    
+    if (coreNumPerTask < 1) {
+      throw new IllegalArgumentException(
+        s"spark.task.cpus is set to an invalid value $coreNumPerTask")
+    }
```

Comment:
we can do that...but is it a good practice to throw exceptions in the constructor?